pkg/manager/config/config_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/pingcap/TiProxy/lib/config"
+	"github.com/stretchr/testify/require"
+)
+
+func waitRetry(t *testing.T, f func(*config.Config) bool, cfgmgr *ConfigManager, msg string) {
+	if f != nil {
+		timer := time.NewTimer(time.Second)
+		for {
+			select {
+			case <-timer.C:
+				t.Fatalf("timeout waiting chan, %s", msg)
+			default:
+				time.Sleep(100 * time.Millisecond)
+				if f(cfgmgr.GetConfig()) {
+					return
+				}
+			}
+		}
+	}
+}
+
+func TestConfig(t *testing.T) {
```

Comment:
Add tests for `Create`, `Remove` and `Rename` events.

Reply:
It doesn't matter, I think. Whatever events you triggered, there will be a reload. The difference is that `remove/rename` needs re-watching.

And it is expected currently removing files longer than 50ms will stop the reloading. Anyway, a event of remove test will be added.